alica_engine/include/engine/AlicaContext.h

```diff
@@ -435,9 +453,21 @@ void AlicaContext::setCommunicator(Args&&... args)
 {
     static_assert(std::is_base_of<IAlicaCommunication, CommunicatorType>::value, "Must be derived from IAlicaCommunication");
 #if (defined __cplusplus && __cplusplus >= 201402L)
-    _communicator = std::make_unique<CommunicatorType>(_engine.get(), std::forward<Args>(args)...);
+    _communicator = std::make_unique<CommunicatorType>(std::bind(&AlicaContext::handleOnSyncTalk, this, std::placeholders::_1),
+            std::bind(&AlicaContext::handleOnSyncReady, this, std::placeholders::_1),
+            std::bind(&AlicaContext::handleIncomingAuthorityMessage, this, std::placeholders::_1),
+            std::bind(&AlicaContext::handlePlanTreeInfo, this, std::placeholders::_1),
+            std::bind(&AlicaContext::handleOnSolverResult, this, std::placeholders::_1),
+            std::bind(&AlicaContext::handleAgentQuery, this, std::placeholders::_1),
+            std::bind(&AlicaContext::handleAgentAnnouncement, this, std::placeholders::_1), std::forward<Args>(args)...);
 #else
-    _communicator = std::unique_ptr<CommunicatorType>(new CommunicatorType(_engine.get(), std::forward<Args>(args)...));
+    _communicator = std::unique_ptr<CommunicatorType>(new CommunicatorType(std::bind(&AlicaContext::handleOnSyncTalk, this, std::placeholders::_1),
```

Comment:
Are we still maintaining old cpp compatibility?  We've been kind of ignoring it in newer code

Reply:
can drop. it was retained earlier for some old alica projects compatibility. Since the api is changed anyway, can drop this and make 17 an officially supported version

Reply:
Can we do this for all places in a separate pr?